api/server/middleware.go

```diff
@@ -28,6 +29,24 @@ func (s *Server) loggingMiddleware(handler httputils.APIFunc) httputils.APIFunc
 	}
 }
 
+// debugRequestMiddleware dumps the request to logger
+// this is implemented separately from `logingMiddleware` so that we don't have to
+// check the logging level or have httputil.DumpRequest called on each request.
+// instead the middleware is only injected when the logging level is set to debug
```

Comment:
I think these last comment lines should go down on line 132 since that's where we do the injection and check the logging level,  right?